docs/Builtin_Tuner.md

```diff
@@ -0,0 +1,313 @@
+
+**Usage example**
+
+```yaml
+# config.yml
+tuner:
+  builtinTunerName: SMAC
+  classArgs:
+    optimize_mode: maximize
+```
+
+<br>
+
+<a name="Batch"></a>
+
+![](https://placehold.it/15/1589F0/000000?text=+) `Batch Tuner`
+
+> Builtin Tuner Name: BatchTuner
+
+**Suggested scenario**
+
+If the configurations you want to try have been decided, you can list them in searchspace file (using `choice`) and run them using batch tuner.
+
+**Usage example**
+
+```yaml
+# config.yml
+tuner:
+  builtinTunerName: BatchTuner
+```
+
+<br>
+
+Note that the search space that BatchTuner supported like:
+
+```json
+{
+    "combine_params":
+    {
+        "_type" : "choice",
+        "_value" : [{"optimizer": "Adam", "learning_rate": 0.00001},
+                    {"optimizer": "Adam", "learning_rate": 0.0001},
+                    {"optimizer": "Adam", "learning_rate": 0.001},
+                    {"optimizer": "SGD", "learning_rate": 0.01}, 
+                    {"optimizer": "SGD", "learning_rate": 0.005}, 
+                    {"optimizer": "SGD", "learning_rate": 0.0002}]
+    }
+}
+```
+
+The search space file including the high-level key `combine_params`. The type of params in search space must be `choice` and the `values` including all the combined-params value.
+
+<a name="GridSearch"></a>
+
+![](https://placehold.it/15/1589F0/000000?text=+) `Grid Search`
+
+> Builtin Tuner Name: **Grid Search**
+
+**Suggested scenario**
+
+Note that the only acceptable types of search space are `choice`, `quniform`, `qloguniform`. **The number `q` in `quniform` and `qloguniform` has special meaning (different from the spec in [search space spec](./SearchSpaceSpec.md)). It means the number of values that will be sampled evenly from the range `low` and `high`.**
+
+It is suggested when search space is small, it is feasible to exhaustively sweeping the whole search space.
+
+**Usage example**
+
+```yaml
+# config.yml
+tuner:
+  builtinTunerName: GridSearch
+```
+
+<br>
+
+<a name="Hyperband"></a>
+
+![](https://placehold.it/15/1589F0/000000?text=+) `Hyperband`
+
+> Builtin Advisor Name: **Hyperband**
+
+**Suggested scenario**
+
+It is suggested when you have limited computation resource but have relatively large search space. It performs good in the scenario that intermediate result (e.g., accuracy) can reflect good or bad of final result (e.g., accuracy) to some extent.
+
+**Requirement of classArg**
+
+* **optimize_mode** (*maximize or minimize, optional, default = maximize*) - If 'maximize', tuners will return the hyperparameter set with larger expectation. If 'minimize', tuner will return the hyperparameter set with smaller expectation.
+* **R** (*int, optional, default = 60*) - the maximum STEPS (could be the number of mini-batches or epochs) can be allocated to a trial. Each trial should use STEPS to control how long it runs.
+* **eta** (*int, optional, default = 3*) - proportion of discarded trials
```

Comment:
I do not understand,  default is 3,  how 3 can be a proportion of discarded trials?

Reply:
@QuanluZhang